### DISCUSSION TITLE
Re-installs Dependencies

### DISCUSSION THREAD

**Discussion by ehsung** [metagenome-atlas/atlas#715] [Q&A]
Hello, I am new to running atlas and didn't find any solid answers pertaining to atlas re-installing its dependencies/conda_envs with every new run.

I successfully completed running the example in the tutorial and specified my atlas_db location. But when I started a new run, using the same atlas_db location, it started re-installing the conda_envs. I also noticed that all of the conda enviroments have tmp names like: 11d3e26a7b2f7d2eec96308c69bda671_ or 1dbbb7e0d0a6d0d3253bfe588a840483_.

Here is an image of my altas_db:
<img width="1422" alt="Screenshot 2024-03-04 at 12 08 01 PM" src="https://github.com/metagenome-atlas/atlas/assets/11777528/53227595-918c-4aa2-8feb-1e43f21633f4">

Is the intentional and what it means by "all databases and other dependencies are installed on the fly"? Is there a way to specify re-using the already installed conda_envs?

**Comment by ehsung** [comment]
I apologize, it looks like there was a minor issue in my setup, that cause the environments to keep re-install, because it wasn't finding it initially. It does know which environments it has already created and uses it as necessary. This is a really cool pipeline, thank you!

**Comment by SilasK** [comment]
Thank you. 
The special names is a way to make an unique identifier for each environment. 
In theory atlas should always reuse the same databases (e.g. taxonomy db) and conda environments. 

You might see new conda install for single environnements if there is a new update to atlas or you use tools you didn't use before. But in theory never a re install of all environments. 

Happy coding.
